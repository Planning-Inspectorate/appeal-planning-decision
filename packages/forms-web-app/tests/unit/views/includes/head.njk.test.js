const nunjucksTestRenderer = require('../nunjucks-render-helper');

describe('views/includes/head', () => {
  describe('Google Analytics section', () => {
    beforeEach(() => {
      ['cookies', 'googleAnalyticsId'].forEach((key) => {
        delete nunjucksTestRenderer.globals[key];
        expect(() => nunjucksTestRenderer.getGlobal(key)).toThrow(`global not found: ${key}`);
      });
    });

    const matchesSnapshop = () =>
      expect(
        nunjucksTestRenderer.renderString(`
    {% include "includes/head.njk" %}
    `)
      ).toMatchSnapshot();

    it(`should not render if googleAnalyticsId and cookies.cookie_policy.usage are not set`, () => {
      matchesSnapshop();
    });

    it(`should not render if googleAnalyticsId is set but cookies.cookie_policy.usage is not set`, () => {
      nunjucksTestRenderer.addGlobal('googleAnalyticsId', 123);
      matchesSnapshop();
    });

    it(`should not render if googleAnalyticsId is not set but cookies.cookie_policy.usage is set`, () => {
      nunjucksTestRenderer.addGlobal('cookies', { cookie_policy: { usage: true } });
      matchesSnapshop();
    });

    it(`should render if googleAnalyticsId and cookies.cookie_policy.usage are set`, () => {
      nunjucksTestRenderer.addGlobal('googleAnalyticsId', 123);
      nunjucksTestRenderer.addGlobal('cookies', { cookie_policy: { usage: true } });
      matchesSnapshop();
    });
  });
});
