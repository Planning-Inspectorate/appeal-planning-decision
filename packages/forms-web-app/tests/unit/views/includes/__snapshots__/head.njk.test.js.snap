// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/includes/head Google Analytics section should not render if googleAnalyticsId and cookies.cookie_policy.usage are not set 1`] = `
"
    

  <!--[if lte IE 8]><link href=\\"/public/stylesheets/main-ie8.css\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><![endif]-->
  <!--[if gt IE 8]><!--><link href=\\"/public/stylesheets/main.css\\" media=\\"all\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><!--<![endif]-->

  <p hidden id=\\"gaId\\"></p>

  

  <script src=\\"/public/common/javascripts/index.bundle.js\\" defer></script>



    "
`;

exports[`views/includes/head Google Analytics section should not render if googleAnalyticsId is not set but cookies.cookie_policy.usage is set 1`] = `
"
    

  <!--[if lte IE 8]><link href=\\"/public/stylesheets/main-ie8.css\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><![endif]-->
  <!--[if gt IE 8]><!--><link href=\\"/public/stylesheets/main.css\\" media=\\"all\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><!--<![endif]-->

  <p hidden id=\\"gaId\\"></p>

  

  <script src=\\"/public/common/javascripts/index.bundle.js\\" defer></script>



    "
`;

exports[`views/includes/head Google Analytics section should not render if googleAnalyticsId is set but cookies.cookie_policy.usage is not set 1`] = `
"
    

  <!--[if lte IE 8]><link href=\\"/public/stylesheets/main-ie8.css\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><![endif]-->
  <!--[if gt IE 8]><!--><link href=\\"/public/stylesheets/main.css\\" media=\\"all\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><!--<![endif]-->

  <p hidden id=\\"gaId\\">123</p>

  

  <script src=\\"/public/common/javascripts/index.bundle.js\\" defer></script>



    "
`;

exports[`views/includes/head Google Analytics section should render if googleAnalyticsId and cookies.cookie_policy.usage are set 1`] = `
"
    

  <!--[if lte IE 8]><link href=\\"/public/stylesheets/main-ie8.css\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><![endif]-->
  <!--[if gt IE 8]><!--><link href=\\"/public/stylesheets/main.css\\" media=\\"all\\" rel=\\"stylesheet\\" type=\\"text/css\\" /><!--<![endif]-->

  <p hidden id=\\"gaId\\">123</p>

  
<!-- Global site tag (gtag.js) - Google Analytics -->

  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', '123');
  </script>
  

  <script src=\\"/public/common/javascripts/index.bundle.js\\" defer></script>



    "
`;
